ddl/ddl_api.go

```diff
@@ -2108,20 +2106,27 @@ func (d *ddl) CoalescePartitions(ctx sessionctx.Context, ident ast.Ident, spec *
 		return errors.Trace(ErrPartitionMgmtOnNonpartitioned)
 	}
 
+	switch meta.Partition.Type {
```

Comment:
This `switch` looks weird.

Reply:
@crazycs520  What if we move the `default:` to the end?

Reply:
How about remove `default` or move the `default:` to the end?